### PR TITLE
feat: better swapper error handling

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Greatly improves quote error handling in preparation for #5829 where we need to track missing trade pairs:
- Adds more granular error tracking
- Renders almost all errors in the UI, except for the few that are unknown to us
- Cleans up most of the tech debt relating to swapper errors

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

In preparation for #5829

## Risk

Moderate risk of broken trades or incorrect error handling.

## Testing

- Check quotes are working
- Check errors are correctly rendered in UI
- Check trades still work

## Screenshots (if applicable)

Most cases look similar to this, where there are multiple successful quotes, followed by a few sad ones:
image

Some examples of more error messages
image
image
image

### CODE REVIEW

**Comment by woodenfurniture** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/shapeshift/web/5991?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `develop`
  * **PR #5971** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5971?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5991** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5991?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5991?utm_source=stack-comment).

**Review by 0xApotheosis — APPROVED** [review]
LGTM.

With this all in, we should probably now make our errors more consistent across swappers. E.g. sometimes swappers show when an asset pair is not supported, other times they don't.

We should also now allow account selection all the time, since we'll tell the user when a swapper doesn't support cross-account.

**Comment by woodenfurniture** [comment]
Note ops has completed pre-merge testing here
https://discord.com/channels/554694662431178782/876953436908822568/1196203757940256918

File: src/lib/swapper/swappers/ThorchainSwapper/cosmossdk/getCosmosTxData.ts
[the remark this answers is not in the document]
Comment (woodenfurniture) [reply src/lib/swapper/swappers/ThorchainSwapper/cosmossdk/getCosmosTxData.ts:1]: yeah it is, not used anywhere and cleaned up up as part of this

**Pull request merged**